pennylane/interfaces/torch.py

```diff
@@ -133,16 +149,9 @@ def forward(ctx, kwargs, *parameters):  # pylint: disable=arguments-differ
             )
 
         ctx.tapes = kwargs["tapes"]
-        ctx.device = kwargs["device"]
+        ctx.jpc = kwargs["jpc"]
 
-        ctx.execute_fn = kwargs["execute_fn"]
-        ctx.gradient_fn = kwargs["gradient_fn"]
-
-        ctx.gradient_kwargs = kwargs["gradient_kwargs"]
-        ctx.max_diff = kwargs["max_diff"]
-        ctx._n = kwargs.get("_n", 1)
-
-        res, ctx.jacs = ctx.execute_fn(ctx.tapes, **ctx.gradient_kwargs)
+        res = tuple(kwargs["execute_fn"](ctx.tapes))

```

Comment:
i'm a little mystified by how this works. where does `ctx.jacs` go? is `res` now both? we directly referenced `ctx.jacs` later in the old code, so I'm curious how it's used now. also not sure where the `_jac_to_torch` logic now lives.

Reply:
We used to cache `jacs` on the context for the backwards pass. Now when we need to cache for the backward pass, `execute_fn` is `DeviceDerivatives.execute_and_cache_jacobian`, and then `DeviceDerivatives` caches the jacobian between execution and the vjp calculation.

So previously the call signature for `execute_fn` was:
```
execute_fn(tapes, **gradient_kwargs) -> (ResultBatch, Optional[Jac]):
```
Where jac could be a jacobian, an empty tuple, or `None`.

Now its simply:
```
execute_fn(tapes) -> ResultBatch
```

But `execute_fn` can have the side effect of computing and caching the jacobian.

Reply:
that makes sense. it's pretty clear what `execute_and_cache_jacobian` does 😄